### DISCUSSION TITLE
What Are the Different Types of Machine Learning?

### DISCUSSION THREAD

**Discussion by luffy-Monkey123** [luffy-Monkey123/python-cool-programs#13] [Q&A] [answered]
What Are the Different Types of Machine Learning?

**Comment by RUTAGENGWA-ASANTE-BRUCE** [comment, marked as answer]
There are three types of machine learning:

Supervised Learning
In supervised machine learning, a model makes predictions or decisions based on past or labeled data. Labeled data refers to sets of data that are given tags or labels, and thus made more meaningful.

Supervised Learning

Unsupervised Learning
In unsupervised learning, we don't have labeled data. A model can identify patterns, anomalies, and relationships in the input data.

Unsupervised Learning

Reinforcement Learning
Using [reinforcement learning](https://www.simplilearn.com/tutorials/machine-learning-tutorial/reinforcement-learning), the model can learn based on the rewards it received for its previous action.

Reinforcement Learning

Consider an environment where an agent is working. The agent is given a target to achieve. Every time the agent takes some action toward the target, it is given positive feedback. And, if the action taken is going away from the goal, the agent is given negative feedback. 

Also Read: [Supervised and Unsupervised Learning in Machine Learning](https://www.simplilearn.com/tutorials/machine-learning-tutorial/supervised-and-unsupervised-learning)
